prover/crates/bin/prover_autoscaler/src/global/scaler.rs

```diff
@@ -182,9 +213,29 @@ impl<K: Key> Scaler<K> {
                                 .cluster_priorities
                                 .get(&b.name)
                                 .unwrap_or(&u32::MAX),
-                        ),
-                ) // Sort by priority.
-                .then(b.max_pool_size.cmp(&a.max_pool_size)) // Reverse sort by cluster size.
+                        )
+                }
+            }
+        };
+
+        pools.sort_by(|a, b| {
+            if self.target_priority.is_some() {
+                // Use target_priority for sorting.
+                std::cmp::Ordering::Equal
```

Comment:
maybe missing something obvious, but why is this equal here?

Reply:
I mean why we are not sorting here by something

Reply:
Key is GPU or nothing, if there is `target_priority`, it should be used for GPU sorting too, so disabling sorting by GPU.